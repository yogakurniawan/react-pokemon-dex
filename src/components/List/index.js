import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'react-flexbox-grid';
import Card from 'components/Card';
import uuid from 'uuid/v1';

function List(props) {
  return (
    <Row center="xs">
      {
        props.items.map((item, index) => (
          <Col key={uuid()} xs={12} sm={6} md={4} lg={3}>
            <Card pokemon={item.pokemon ? item.pokemon : item} showDetail={props.showDetail} />
          </Col>
        ))
      }
    </Row>
  );
}

List.propTypes = {
  showDetail: PropTypes.func,
  items: PropTypes.array,
};

export default List;
