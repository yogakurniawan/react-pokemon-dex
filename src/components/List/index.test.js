import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import List from './index';

injectTapEventPlugin();

const props = {
  items: [],
  showDetail: jest.fn()
}

describe('List', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <List {...props} />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});