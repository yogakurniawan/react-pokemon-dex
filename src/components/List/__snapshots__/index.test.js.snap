// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div
  className="row center-xs"
/>
`;
